app/frontend/src/AppRoute.tsx

```diff
@@ -46,6 +47,8 @@ export default function AppRoute({
 
   const classes = useStyles();
 
+  const RouteContent = () => <ErrorBoundary>{children}</ErrorBoundary>;
```

Comment:
Sorry, I'm an idiot, components in a render function do need to be memoized. Better still would be moving it into a separate function. That way it doesn't get recreated and also doesn't have the (small) overhead of memoizing.

Reply:
Alternatively, I'd also consider moving the error boundary to `App.tsx`, plus that way it will help in case there's an error in the navigation or theme.

Reply:
Not sure if we need to move the ErrorBoundary from AppRoute to App level. Instead we can add an additional ErrorBoundary to App.

When a Route throws an error, it will still show the top navigation bar and the error will be reset when user navigates to another page. While if we just have a root ErrorBoundary, all the user will see is the ErrorFallback component.

Does that makes sense?

Reply:
Yes good point